guardian/actions-riff-raff
# fix(esbuild): Include packages in bundle

## What does this change?
Fixes a regression introduced in https://github.com/guardian/actions-riff-raff/pull/152 (and [v4.0.4](https://github.com/guardian/actions-riff-raff/releases/tag/v4.0.4)).

I hadn't read the [release notes](https://github.com/evanw/esbuild/releases/tag/v0.22.0) carefully enough, and missed that v0.22.0 of esbuild was breaking with regards to the inclusion of packages in the bundle.

> This breaking change is an experiment. People are commonly confused when using esbuild to bundle code for node (i.e. for --platform=node) because some packages may not be intended for bundlers, and may use node-specific features that don't work with a bundler. Even though esbuild's "getting started" instructions say to use --packages=external to work around this problem, many people don't read the documentation and don't do this, and are then confused when it doesn't work. So arguably this is a bad default behavior for esbuild to have if people keep tripping over this.
>
> With this release, esbuild will now omit packages from the bundle by default when the platform is node (i.e. the previous behavior of --packages=external is now the default in this case). Note that your dependencies must now be present on the file system when your bundle is run. If you don't want this behavior, you can do --packages=bundle to allow packages to be included in the bundle (i.e. the previous default behavior). Note that --packages=bundle doesn't mean all packages are bundled, just that packages are allowed to be bundled. You can still exclude individual packages from the bundle using --external: even when --packages=bundle is present.

Set the `--packages=bundle` flag to include all packages in the bundle.

In hindsight, the diff to `dist/index.js` in #152 should have indicated an issue:

![image](https://github.com/guardian/actions-riff-raff/assets/836140/d77b8176-c0e5-455f-ae90-2683187f9819)

## How to test
Use this action from the latest commit on this branch. See https://github.com/guardian/service-catalogue/pull/1144.
